### DISCUSSION TITLE
What's the different between Nuxt Axios & serverMiddleware

### DISCUSSION THREAD

**Discussion by codihaus** [nuxt/nuxt#9203] [Questions (Nuxt 2)] [answered]
Hello,

We're new to NuxtJS, and I have a question about Server Side Render.

We want to create SSR website for SEO, but we're confused of using Nuxt Axios vs using serverMiddleware (/api folder i.e inside the nuxtjs project)

Will serverMiddleware be best for SSR instead of Axios? and what's your favourite method on this case?

Thank you!

**Comment by danielroe** [comment, marked as answer]
`@nuxtjs/axios` is a way of performing requests from your client-side Nuxt app. `serverMiddleware` is a way of adding request handlers on the server-side that respond to URLs. So they aren't the same thing. In fact, you could use axios to perform a request to an endpoint that's handled by your server middleware.

Relevant documentation:
* https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-servermiddleware
* https://axios.nuxtjs.org/
